### PR TITLE
Refactor `/account` routes

### LINKED ISSUE #867: Reconsider routing for `/account` page
For scalability purposes, we could reconsider the way we handle routes, in regards to the user `/account` page.

Currently, all views under the user account are based on an _active_ section `hash`, as seen [here](https://github.com/Qiskit/platypus/blob/main/frontend/vue/components/UserAccount/UserAccountLayout.vue#L44-L52). However, with this approach, we run into an issue with scaling — for example, trying to implement something like `account > classroom > syllabus > create`... or `account > classroom > syllabus > edit`.

This issue is to propose changing this strategy from using the _sectionId_ approach, in favor of a URL-based approach(?) so that we have routes such as 

- `/account/mylearning`
- `/account/classroom`
- `/account/classroom/syllabus/create`
- `/account/classroom/syllabus/edit`

### PR DESCRIPTION
**Pull request by techtolentino**
## Changes

Fixes #867 

## Implementation details
- add new routes for `/account/classroom`, `/account/privacy`
- replaced `hash` based approach, with `pathname` for `activeSection` matching
- replaced `AppLink` w/ `BasicLink` for account sidebar

## How to read this PR
- review `getAccountData` in `app.ts` to see if this is the right approach
- review UI (or screen grab below)

## Screenshots

https://user-images.githubusercontent.com/6276074/164274419-1c464477-2e55-454c-a4ad-c47dc164780c.mov

## Notes
- I'm making this PR into the `feature/professor-syllabus` branch, but please let me know if this should come _after_ that branch is merged. In other words, should this be part of a later Syllabus 1.1 milestone?
- This PR does **not** handle the `/syllabus` related routes...  I think that needs to be considered in separate refactor, along with [this issue](https://github.com/Qiskit/platypus/issues/837)

### CODE REVIEW

File: server/app.ts [hunk 1]
```diff
@@ -84,6 +84,36 @@ const getUserProgressData = async (req: Request) => {
...
+    return res.redirect('/eula')
+  }
+
+  const lang = req.locale.id || 'en'
+  const translationsJSON = JSON.stringify(TRANSLATIONS[lang] || {})
+
+  const privacyPolicyMD = loadLocaleRawFile('privacy-policy.md', lang)
+
+  const userMockData = {
+    firstName: req.user?.firstName,
```
Comment (Tansito) [inline server/app.ts:101]: How we are checking if the `user` exists in the first line of the method I think we can assume that the user exists at this point and remove the `?` after `req.user`.

**Review by Tansito — CHANGES_REQUESTED** [review]

**Comment by Tansito** [comment]
Thanks @techtolentino ! I just commented a little change that I saw but apart from that for me it's ok, do you think that for syllabus the routes will need a refactor? We can try to do it for Syllabus 1.1.

File: server/app.ts [hunk 2]
```diff
@@ -84,6 +84,36 @@ const getUserProgressData = async (req: Request) => {
   return progress
 }
 
+const getAccountData = async (req: Request, res: Response) => {
+  if (!req.user) {
+    return res.redirect('/signin')
+  }
+  if (req.user && !req.user.acceptedPolicies) {
+    return res.redirect('/eula')
+  }
```
Comment (vabarbosa) [inline server/app.ts:93]: along the lines of what @Tansito mentioned, at this point we already know we have `req.user` (checked in statement above this one) so no need to check it here

```suggestion
  if (!req.user.acceptedPolicies) {
    return res.redirect('/eula')
  }
```

File: server/app.ts [hunk 3]
```diff
@@ -84,6 +84,36 @@ const getUserProgressData = async (req: Request) => {
...
+
+  const lang = req.locale.id || 'en'
+  const translationsJSON = JSON.stringify(TRANSLATIONS[lang] || {})
+
+  const privacyPolicyMD = loadLocaleRawFile('privacy-policy.md', lang)
+
+  const userMockData = {
+    firstName: req.user.firstName,
+    lastName: req.user.lastName
+  }
```
Comment (vabarbosa) [inline server/app.ts:103]: does `userMockData` still make sense here as a variable name? maybe `loggedInUser` or just `userData`

**Review by vabarbosa — APPROVED** [review]
this is great @techtolentino! just had a couple very minor suggestions

File: server/app.ts (hunk 3 above)
Comment (vabarbosa) [reply server/app.ts]: thanks looks good 👍🏽

**Pull request merged**